src/riak_ql_ddl.erl

```diff
@@ -1017,6 +1095,100 @@ is_insert_valid_test_helper(Table_name, Table_def, Insert) ->
         "    myfamily, myseries, time))"
        ).
 
+insert_sql_columns_empty_fields_test() ->
+    {_DDL, Mod} = is_sql_valid_test_helper("mytab", ?LARGE_TABLE_DEF),
+    FInMod = [ {identifier,[F]} || {[F], _I} <- Mod:get_field_positions() ],
+    FieldsIn = [],
+    ValuesIn = [[{varchar, <<"family1">>},
+            {varchar, <<"series1">>},
+            {timestamp, 1000},
+            {varchar, <<"sunny">>},
+            {double, 37.0}]],
+    {FieldsOut, ValuesOut } = insert_sql_columns(Mod, FieldsIn, ValuesIn),
+    ?assertEqual(FInMod, FieldsOut),
+    ?assertEqual(ValuesIn, ValuesOut).
+
+insert_sql_columns_pk_fields_test() ->
+    {_DDL, Mod} = is_sql_valid_test_helper("mytab", ?LARGE_TABLE_DEF),
+    FInMod = [ {identifier,[F]} || {[F], _I} <- Mod:get_field_positions() ],
+    FieldsIn = lists:sublist(FInMod, 3),
+    ValuesIn = [[{varchar, <<"family1">>},
+            {varchar, <<"series1">>},
+            {timestamp, 1000}]],
+    {FieldsOut, ValuesOut } = insert_sql_columns(Mod, FieldsIn, ValuesIn),
+    %% expecting values specified plus nulls
+    ValuesExpected = [hd(ValuesIn) ++
+        lists:sublist(hd(ValuesOut), length(FieldsIn) + 1, length(FInMod) - length(FieldsIn))],
+    ?assertEqual(FInMod, FieldsOut),
+    ?assertEqual(ValuesExpected, ValuesOut).
+
+insert_sql_columns_all_fields_test() ->
+    {_DDL, Mod} = is_sql_valid_test_helper("mytab", ?LARGE_TABLE_DEF),
+    FInMod = [ {identifier,[F]} || {[F], _I} <- Mod:get_field_positions() ],
+    FieldsIn = FInMod,
+    ValuesIn = [[{varchar, <<"family1">>},
+            {varchar, <<"series1">>},
+            {timestamp, 1000},
+            {varchar, <<"sunny">>},
+            {double, 37.0}]],
+    {FieldsOut, ValuesOut} = insert_sql_columns(Mod, FieldsIn, ValuesIn),
+    ?assertEqual(FInMod, FieldsOut),
+    ?assertEqual(ValuesIn, ValuesOut).
+
+%% Shuffle two equal length lists equivalently
+shuffle2(L0, L1) ->
+    shuffle2(list_to_tuple(L0), list_to_tuple(L1), length(L0)).
+
+shuffle2(L0, L1, 0) ->
+    {tuple_to_list(L0), tuple_to_list(L1)};
+shuffle2(L0, L1, Len) ->
+    Rand = random:uniform(Len),
```

Comment:
All but one case of the shuffle covers the case of the insert statement handling columns specified out of order. Hence, going whole hog is over coverage of the feature. The one case that fails to provide coverage is when the shuffle results in the column order being mapped back to the exact same column order.

I considered swapping the first and last columns to achieve this result, but felt that the odds of an exact order match were low enough to have this more random method of coverage provide the coverage.

I also considered using function guards:
* when the resulting list is empty (entering the recursion), start the shuffle
* when the resulting list matches the input list, restart the shuffle
* when the resulting list matches the length of the input list (but not the exact content), return

This seems like overkill.